docs/reference/python/conventions.md

```diff
@@ -36,14 +36,98 @@ and future members of the software team. The major things that we document in ou
 Ideally, the third point should be avoided as much as possible since we would want our code to be
 self explanatory. It should be done only when absolutely necessary.
 
+### Type hinting and static type checking
+
+Since Python is a dynamically typed language, we need to make use of the [`typing`](https://docs.python.org/3/library/typing.html){target=_blank}
+and [`mypy`](https://mypy.readthedocs.io/en/stable/index.html){target=_blank} modules if we want to do
+any type checking in our functions and classes. Type checking is beneficial for documentation and maintaining a clean
+software architecture. There is some syntax to get familiar in order to use type checking. We recommend the following
```

Comment:
Explain why we type hint first
- helps catch errors through static type checking
    - mypy is a static type checker